lib/logstash/inputs/dynamodb.rb

```diff
@@ -0,0 +1,359 @@
+# encoding: utf-8
+#
+#Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+#Licensed under the Apache License, Version 2.0 (the "License");
+#you may not use this file except in compliance with the License.
+#You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#Unless required by applicable law or agreed to in writing, software
+#distributed under the License is distributed on an "AS IS" BASIS,
+#WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#See the License for the specific language governing permissions and
+#limitations under the License.
+#
+require "logstash/inputs/base"
+require "logstash/namespace"
+require "securerandom"
+require "thread"
+require "socket"
+require_relative "LogStashRecordProcessorFactory"
+require_relative "DynamoDBLogParser"
+
+#Java Dependencies: KCL, Streams Adapter, Log4j, Preview AWS SDK
+require 'java'
+begin
+  require 'jar-dependencies'
+  require_jar( 'com.amazonaws', 'amazon-kinesis-client', '1.6.0' )
+  require_jar( 'log4j', 'log4j', '1.2.17' )
+  require_jar( 'com.google.guava', 'guava', '15.0' )
+  require_jar( 'com.amazonaws', 'aws-java-sdk-core', '1.10.10' )
+  require_jar( 'com.amazonaws', 'aws-java-sdk-cloudwatch', '1.10.8' )
+  require_jar( 'com.amazonaws', 'aws-java-sdk-dynamodb', '1.10.10' )
+  require_jar( 'com.amazonaws', 'aws-java-sdk-kinesis', '1.10.8' )
+  require_jar( 'com.amazonaws', 'dynamodb-streams-kinesis-adapter', '1.0.0' )
+  require_jar( 'com.amazonaws', 'dynamodb-import-export-tool', '1.0.0' )
+  require_jar( 'commons-logging', 'commons-logging', '1.1.3')
+  require_jar( 'org.apache.httpcomponents', 'httpclient', '4.3.6')
+  require_jar( 'org.apache.httpcomponents', 'httpcore', '4.3.3')
+  require_jar( 'com.fasterxml.jackson.core', 'jackson-databind', '2.5.3' )
+  require_jar( 'joda-time', 'joda-time', '2.8.1')
+end
+
+java_import "com.amazonaws.AmazonClientException"
+java_import "org.apache.log4j.LogManager"
+java_import "org.apache.log4j.Level"
+java_import "com.fasterxml.jackson.annotation.JsonInclude"
+java_import "com.amazonaws.regions.RegionUtils"
+
+module AmazonDynamoDB
+  include_package "com.amazonaws.services.dynamodbv2"
+  include_package "com.amazonaws.services.dynamodbv2.streamsadapter"
+  include_package "com.amazonaws.services.dynamodbv2.model"
+end
+module AmazonCredentials
+  include_package "com.amazonaws.auth"
+  include_package "com.amazonaws.internal"
+end
+
+module DynamoDBBootstrap
+  include_package "com.amazonaws.dynamodb.bootstrap"
+end
+
+module CloudWatch
+  include_package "com.amazonaws.services.cloudwatch"
+end
+
+module KCL
+  include_package "com.amazonaws.services.kinesis.clientlibrary.lib.worker"
+end
+
+#DynamoDBStreams plugin that will first scan the DynamoDB table
+#and then consume streams and push those records into Logstash
+class LogStash::Inputs::DynamoDB < LogStash::Inputs::Base
+  config_name "dynamodb"
+
+  LF_DYNAMODB = "dymamodb"
```

Comment:
I know is not a common ruby practice in some places, but using ```"foobar".freeze``` for constance is important, so then constance are really what they are immutable things.